lib/ansible/modules/cloud/amazon/ec2_vpc_net_facts.py

```diff
@@ -57,76 +60,182 @@
+                    returned: always
+                    type: dict
+                    contains:
+                        state:
+                            description: The CIDR block association state.
+                            returned: always
+                            type: string
+'''
 
+import traceback
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import connect_to_aws, ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    ec2_argument_spec,
+    get_aws_connection_info,
+    AWSRetry,
+    HAS_BOTO3,
+    boto3_tag_list_to_ansible_dict,
+    camel_dict_to_snake_dict,
+    ansible_dict_to_boto3_filter_list
+)
 
+try:
+    from botocore.exceptions import ClientError, NoCredentialsError
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
 
-def get_vpc_info(vpc):
 
-    try:
-        classic_link = vpc.classic_link_enabled
-    except AttributeError:
-        classic_link = False
+@AWSRetry.backoff()
+def describe_vpcs(connection, module):
+    """
+    Describe VPCs.
 
-    vpc_info = { 'id': vpc.id,
-                 'instance_tenancy': vpc.instance_tenancy,
-                 'classic_link_enabled': classic_link,
-                 'dhcp_options_id': vpc.dhcp_options_id,
-                 'state': vpc.state,
-                 'is_default': vpc.is_default,
-                 'cidr_block': vpc.cidr_block,
-                 'tags': vpc.tags
-               }
+    connection  : boto3 client connection object
+    module  : AnsibleModule object
+    """
+    # collect parameters
+    filters = ansible_dict_to_boto3_filter_list(module.params.get('filters'))
+    vpc_ids = module.params.get('vpc_ids')
 
-    return vpc_info
+    # init empty list for return vars
+    vpc_info = list()
+    vpc_list = list()
 
-def list_ec2_vpcs(connection, module):
+    # Get the basic VPC info
+    try:
+        response = connection.describe_vpcs(VpcIds=vpc_ids, Filters=filters)
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
 
-    filters = module.params.get("filters")
-    vpc_dict_array = []
+    # Loop through results and create a list of VPC IDs
+    for vpc in response['Vpcs']:
+        vpc_list.append(vpc['VpcId'])
 
+    # We can get these results in bulk but still needs two separate calls to the API
     try:
-        all_vpcs = connection.get_all_vpcs(filters=filters)
-    except BotoServerError as e:
-        module.fail_json(msg=e.message)
+        cl_enabled = connection.describe_vpc_classic_link(VpcIds=vpc_list)
+        cl_dns_support = connection.describe_vpc_classic_link_dns_support(VpcIds=vpc_list)
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
```

Comment:
Same here. Also, the individual `describes` should be in separate retries from the list describe, since a throttle here would cause the whole function to re-execute. Really we'd want an individual throttle to just keep retrying that call, not start over completely.